subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -180,13 +184,24 @@ protected DefaultToolingModelBuilderRegistry decorateToolingModelRegistry(Defaul
         return buildScopedToolingModelBuilders.createChild();
     }
 
-    protected PluginManagerInternal createPluginManager(Instantiator instantiator, InstantiatorFactory instantiatorFactory, BuildOperationRunner buildOperationRunner, UserCodeApplicationContext userCodeApplicationContext, CollectionCallbackActionDecorator decorator, DomainObjectCollectionFactory domainObjectCollectionFactory) {
-        PluginTarget target = new RuleBasedPluginTarget(
+    protected PluginManagerInternal createPluginManager(
+        Instantiator instantiator,
+        InstantiatorFactory instantiatorFactory,
+        BuildOperationRunner buildOperationRunner,
+        UserCodeApplicationContext userCodeApplicationContext,
+        CollectionCallbackActionDecorator decorator,
+        DomainObjectCollectionFactory domainObjectCollectionFactory,
+        PluginScheme pluginScheme,
+        SoftwareTypeRegistry softwareTypeRegistry
+        ) {
+        PluginTarget ruleBasedTarget = new RuleBasedPluginTarget(
             project,
+            new ImperativeOnlyPluginTarget<>(project),
             get(ModelRuleExtractor.class),
             get(ModelRuleSourceDetector.class)
         );
-        return instantiator.newInstance(DefaultPluginManager.class, get(PluginRegistry.class), new PluginInstantiator(instantiatorFactory, this), target, buildOperationRunner, userCodeApplicationContext, decorator, domainObjectCollectionFactory);
+        PluginTarget target = new AddSoftwareTypesAsExtensionsPluginTarget(project, ruleBasedTarget, pluginScheme.getInspectionScheme(), softwareTypeRegistry);
```

Comment:
💭 Should the `instantiator` be used to create this instance as well?

Reply:
I suppose it could, but it isn't injecting anything or require any decoration, so I'm not sure what the value it would offer.  In the absence of a compelling reason, I'd just leave it as is.

Reply:
Sounds good.  Not relevant to this PR, but what do you think about having `instantiator` throw an exception if it is used to create a type that isn't `abstract` and lacks any `@Inject`s?  For instance `DefaultPluginManager` right below seems like it could be constructed normally?  

Sometimes I'm confused about if an `instantiator` is really needed, and it seems like it would always be faster to call the constructor if possible.

Reply:
Not sure why PluginManager is instantiated here.  Certainly looks like it could just be directly instantiated.  I wonder if it has to do with the implicit Closure-Action decoration or something like that?  I wouldn't be opposed to making this stricter to avoid unnecessary instantiation, but I think we'd need to do some research to capture all of the reasons why things might be instantiated - I suspect there might be some subtle ones that we're not thinking about.

Reply:
> I suspect there might be some subtle ones that we're not thinking about.

Seems likely.  I might try and experiment with this and see what breaks.